src/main/java/com/musala/atmosphere/agent/AgentManager.java

```diff
@@ -162,37 +127,52 @@ public String createAndStartEmulator(EmulatorParameters parameters) throws Remot
         return emulatorName;
     }
 
-    @Override
+    /**
+     * Closes the process of an emulator specified by it's serial number.
+     *
+     * @param serialNumber
+     *        Serial number of the emulator we want closed.
+     *
+     * @throws DeviceNotFoundException
+     *         Thrown when a method an Agent method was invoked with serial number of a device that is not present on
+     *         the Agent.
+     * @throws IOException
+     *         thrown when an I/O exception of some sort has occurred.
+     * @throws NotPossibleForDeviceException
+     *         thrown when a command for real devices only was attempted on an emulator and vice versa.
+     *
+     */
     public void closeAndEraseEmulator(String serialNumber)
-        throws DeviceNotFoundException,
+            throws DeviceNotFoundException,
             NotPossibleForDeviceException,
             IOException {
         emulatorManager.closeAndEraseEmulator(serialNumber);
     }
 
-    @Override
-    public String getAgentId() throws RemoteException {
+    /**
+     * Gets the unique identifier of the current Agent.
+     *
+     * @return Unique identifier for the current Agent.
+     */
+    public String getAgentId() {
         return agentId;
     }
 
-    @Override
-    public String getInvokerIpAddress() throws RemoteException {
-        try {
-            return RemoteServer.getClientHost();
-        } catch (ServerNotActiveException e) {
-            // Thrown when this method is not invoked by RMI. Nothing to do here, this should not happen.
-            LOGGER.warn("The getInvokerIpAddress method was invoked locally and resulted in exception.");
-        }
-        return "";
-    }
-
-    @Override
-    public void registerServer(String serverIPAddress, int serverRmiPort) throws RemoteException {
+    /**
+     * Registers the server for events, related to changes in the devices on the agent. Should be called after the
+     * server has published it's IAgentEventSender.
+     *
+     * @param serverIpAddress
+     *        The server's IP address.
+     * @param serverPort
+     *        Port on which the WebSocket connection is opened.
+     */
+    public void registerServer(String serverIpAddress, int serverPort) {
```

Comment:
this is confusing. Why do you pass in parameters that you do not use anywhere but in logging?